lib/widgets/progress.ts

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2016 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as _ from 'lodash';
+import ProgressBarFormatter from 'progress-bar-formatter';
+
+const CARRIAGE_RETURN = '\u001b[1A';
+
+const formatDuration = function (seconds: number) {
+	const SECONDS_PER_MINUTE = 60;
+	const SECONDS_PER_HOUR = 3600;
+
+	const hours = Math.floor(seconds / SECONDS_PER_HOUR);
+	seconds %= SECONDS_PER_HOUR;
+
+	const minutes = Math.floor(seconds / SECONDS_PER_MINUTE);
+	seconds = Math.floor(seconds % SECONDS_PER_MINUTE);
+	const padSeconds = seconds < 10 ? '0' + seconds : seconds;
+	const padMinutes = minutes < 10 ? '0' + minutes : minutes;
+
+	if (hours > 0) {
+		return `${hours}h${padMinutes}m${padSeconds}s`;
+	}
+	if (minutes > 0) {
+		return `${minutes}m${padSeconds}s`;
+	}
+	return `${seconds}s`;
+};
+
+export interface ProgressState {
+	percentage: number;
+	message?: string;
+	eta?: number;
+}
+
+class Progress {
+	private _message: string;
+	private _bar: ProgressBarFormatter;
+	private _lastLine: string | undefined;
+
+	/**
+	 * @summary Create a CLI Progress Bar
+	 * @name Progress
+	 * @class
+	 * @public
+	 * @memberof visuals
+	 *
+	 * @param {String} message - message
+	 * @returns {Progress} progress bar instance
+	 *
+	 * @throws Will throw if no message.
+	 *
+	 * @example
+	 * progress = new visuals.Progress('Hello World')
+	 */
+	constructor(message: string) {
+		if (_.trim(message) === '') {
```

Comment:
One less lodash usage
```suggestion
		if (message.trim() === '') {
```